firewood/src/merkle.rs

```diff
@@ -1090,76 +1103,36 @@ impl<S: ShaleStore<Node> + Send + Sync> Merkle<S> {
     where
         K: AsRef<[u8]>,
     {
-        let key_nibbles = Nibbles::<0>::new(key.as_ref());
-
         let mut proofs = HashMap::new();
         if root.is_null() {
             return Ok(Proof(proofs));
         }
 
-        // Skip the sentinel root
-        let root = self
-            .get_node(root)?
-            .inner
-            .as_branch()
-            .ok_or(MerkleError::NotBranchNode)?
-            .children[0];
-        let mut u_ref = match root {
-            Some(root) => self.get_node(root)?,
-            None => return Ok(Proof(proofs)),
```

Comment:
this needs a test case. It's not clear if it's handled correctly with the new code. The case here is that there is a sentinel node but no first child.

Reply:
https://github.com/ava-labs/firewood/pull/380/files#diff-6a6e300f3d9050248dbafa719531f14eabecc1bfac2169bb2527c81ee5474ecaR1841

Reply:
I think that's a test for `range_proof` that does the same thing? I don't believe it gets to `proof` since it can't find the first key. Is that incorrect?

Reply:
bad link... edited